### DISCUSSION TITLE
Color maintenance going from tiff to png

### DISCUSSION THREAD

**Discussion by Jmuccigr** [ImageMagick/ImageMagick#2911] [Help] [answered]
(Seems to a colorspace day)

I'm converting a file I received as an uncompressed TIFF into something compressed. My default is to use png, but when I did that the colors in the TIFF noticeably changed (brightened). The tiff is in CMYK with a profile of Europe ISO Coated FOGRA27. I've tried a few things to get the colors to match, but can't quite figure it out. I don't work with color profiles all that much, so I figured I'd ask to see whether I'm missing something obvious or not-so-obvious.

TIA.

MacOS 10.13.5
Version: ImageMagick 7.0.10-38 Q16 x86_64 2020-11-16 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1)
Delegates (built-in): bzlib fftw fontconfig freetype heic jng jp2 jpeg lcms ltdl lzma png tiff webp wmf x xml zlib

**Comment by snibgo** [comment]
You don't show your command, or input image, or output image. Should we just guess?

**Reply by Jmuccigr** [reply]
> Should we just guess?

No, but you might ask nicely. :-) I was hoping that this was an obvious thing I was missing or ignorant of and didn't require the specific images.

The command that shows the effect is basic: `magick file.tiff file.png`

Then I've tried a bunch of different options like `-colorspace CMYK` (which won't work, I think, because the png format doesn't support that) and exported the profile of the tiff and used it to make the png, all to no change.

I attach a cropped version of the image as a png and a zipped version of the tiff, since GitHub apparently doesn't support that format. I also made a composite image showing both in screen shots. You can see how the png colors are brighter.

Screenshot showing both:
![Both](https://user-images.githubusercontent.com/615115/101040034-030d7e00-354a-11eb-933c-d78e767ccb7c.png)

Crop of the png version:
![file](https://user-images.githubusercontent.com/615115/101034835-980f7780-3548-11eb-9c60-94d71ea1e7a4.png)

[file.tif.zip](https://github.com/ImageMagick/ImageMagick/files/5636654/file.tif.zip)

**Comment by snibgo** [comment, marked as answer]
file.tif is CMYK with an embedded profile. PNG cannot record CMYK, so you should convert it to sRGB, with "-profile":
```
magick file.tif -profile sRGB.icc x.png
```
sRGB.icc is provided with ImageMagick. If sRGB.icc isn't in your current directory, provide the full path. You could use any sRGB ICC profile.

**Comment by Jmuccigr** [comment]
Thanks. That did it.

I do not get the icc files with my install of IM via homebrew. I happen to have them because I've got a local testing server which has IM on it. Is there a preferred place to download them from? I don't see them in this repo.

**Comment by fmw42** [comment]
See profiles at https://imagemagick.org/download/delegates/

**Reply by Jmuccigr** [reply]
Thanks. Is this stuff on GitHub somewhere, so I can easily keep it updated?
